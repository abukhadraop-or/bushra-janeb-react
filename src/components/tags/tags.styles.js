import styled from 'styled-components';
import breakpoints from 'styles/breaking-points';
import colors from 'styles/global-colors';

export const Holder = styled.div`
  background-color: ${colors.concrete};
  border-radius: 0.25rem;
  height: fit-content;
  margin-left: 0;
  margin-top: 1.7rem;
  padding: 0.313rem 0.625rem 0.625rem 0.625rem;
  width: fit-content;

  @media screen and (min-width: ${breakpoints.LargeScreen}) {
    margin-left: 3rem;
  }

  @media screen and (max-width: ${breakpoints.desktopExtraWide}) {
    padding: 0;
  }
`;

export const Title = styled.p`
  display: flex;
  flex-direction: column;
  margin-bottom: 1rem;
  margin-top: 0;
`;

export const Tags = styled.div`
  background-color: ${colors.aluminium};
  border-radius: 10rem;
  color: ${colors.white};
  cursor: pointer;
  font-size: 0.8rem;
  margin-bottom: 0.2rem;
  margin-right: 0.188rem;
  padding: 0.1rem 1rem;
  text-align: center;
  width: fit-content;

  &:hover {
    color: ${colors.darkBlue};
  }
`;

export const TagsContainer = styled.div`
  display: grid;
  grid-template-columns: auto auto auto;
  padding-right: 0.6em;
  width: fit-content;

  @media screen and (max-width: 73.75rem) {
    display: flex;
    flex-wrap: wrap;
    justify-content: space-between;
  }
`;
