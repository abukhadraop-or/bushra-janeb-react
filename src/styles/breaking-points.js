const breakpoints = {
  // 360px
  mobile: '20rem',

  // 480px
  mobileWide: '21.438rem',

  // 650px
  tablet: '54.625rem',

  // 768px
  tabletWide: '48rem',

  // 980px
  desktop: '60rem',

  // 1180px
  desktopWide: '73.75rem',

  // 10444px
  desktopMax: '65.25',

  // 1092x
  desktopExtraWide: '68.25rem',

  // 1180px.
  LargeScreen: '73.75rem',
};

export default breakpoints;
