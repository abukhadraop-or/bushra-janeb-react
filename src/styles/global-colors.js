const colors = {
  black: '#000',
  concrete: '#f3f3f3',
  darkBlue: '#00000066',
  fern: '#5cb85c',
  Gray90: '#E5E5E5',
  lightGrey: '#bbbbbb',
  midnightBlue: '#0000004d',
  transparent: '#ffffff00',
  white: '#ffffff',
  veryLightGrey: '#ddd',
  aliceBlue: '#eceeef',
  aluminium: '#818a91',
};

export default colors;
