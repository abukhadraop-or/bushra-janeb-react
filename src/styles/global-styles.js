import { createGlobalStyle } from 'styled-components';

const GlobalStyle = createGlobalStyle`
  body {
    margin: 0;
    padding: 0;
    font-family: Open-Sans, Helvetica, Sans-Serif;
  }

  *,
  *::before,
  *::after {
    box-sizing: border-box;
  }
`;

export default GlobalStyle;
