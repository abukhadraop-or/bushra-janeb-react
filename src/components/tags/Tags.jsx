import PropTypes from 'prop-types';
import React from 'react';
import { Holder, Tags, TagsContainer, Title } from './tags.styles';

/**
 * Render the <Tag> component.
 *
 * @param {Object} props             Props.
 * @param {Func}   props.onTagSelect Function that handles tag select.
 * @param {Array}  props.tags        The list of tags.
 *
 * @return {JSX.Element}
 */
function Tag({ onTagSelect, tags }) {
  return (
    <Holder>
      <Title> Popular Tags </Title>
      <TagsContainer>
        {tags.map((tag) => (
          <Tags key={tag.id} onClick={() => onTagSelect(tag)}>
            {tag.name}
          </Tags>
        ))}
      </TagsContainer>
    </Holder>
  );
}
Tag.propTypes = {
  onTagSelect: PropTypes.func.isRequired,
  tags: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number,
      name: PropTypes.string,
    })
  ).isRequired,
};

export default Tag;
