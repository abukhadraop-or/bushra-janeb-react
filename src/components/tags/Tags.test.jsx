import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import Enzyme, { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import Tag from './Tags';
import { Tags } from './tags.styles';

Enzyme.configure({ adapter: new Adapter() });

jest.mock('services/fake-tags-service', () => jest.fn());

describe('Tag', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(
      <Tag
        onTagSelect={jest.fn()}
        tags={[
          {
            id: 1,
            name: 'C++',
          },
        ]}
      />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('calls onTagSelect on click on a tag', () => {
    const wrapper = mount(
      <Tag
        onTagSelect={jest.fn()}
        tags={[
          {
            id: 1,
            name: 'C++',
          },
        ]}
      />
    );
    const button = wrapper.find(Tags);
    button.simulate('click');
    expect(wrapper.props().onTagSelect).toHaveBeenCalled();
  });
});
