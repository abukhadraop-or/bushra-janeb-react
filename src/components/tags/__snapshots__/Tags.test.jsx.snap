// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag should match snapshot 1`] = `
<styled.div>
  <styled.p>
     Popular Tags 
  </styled.p>
  <styled.div>
    <styled.div
      key="1"
      onClick={[Function]}
    >
      C++
    </styled.div>
  </styled.div>
</styled.div>
`;
